scripts/send_application_on_closure.py

```diff
@@ -43,13 +47,15 @@ def send_incomplete_applications_after_deadline(fund_id, round_id, send_emails=F
 
     fund_rounds = get_fund_round(fund_id, round_id)
     if current_date_time > fund_rounds.get("deadline"):
+        fund_data = get_fund(fund_id)
         search_params = {
             "status_only": ["NOT_STARTED", "IN_PROGRESS", "COMPLETED"],
             "fund_id": fund_id,
             "round_id": round_id,
+            "application_id": application_id if single_application else None,
```

Comment:
Should probably check if `application_id` is supplied in this method (as well as for the cmd args) as if it's not then I think you'll end up emailing everyone?

Reply:
I see what you mean. I have set --single_application as a required arg so we wont send all applications by mistake

Reply:
I think this will still send 'application_id' as None to the search. Does the search return everything in that instance or nothing?

Reply:
It returns everything. There is already a check in the query function (matching_applications = search_applications(**search_params)

```
  if application_id:
        filters.append(Applications.id == application_id)
    found_apps = get_applications(filters, include_forms=forms, as_json=True)
    return found_apps
```

Reply:
If you are still not convinced. Happy to add the validation here as well 😄

Reply:
Haha no that's fine 👍